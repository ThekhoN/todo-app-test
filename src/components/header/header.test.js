import React from "react";
import { shallow } from "enzyme";
import Header from "./index";

describe("Header", () => {
  let wrapper,
    title = "Hello World!";
  it("matches snapshot", () => {
    wrapper = shallow(<Header title={title} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly", () => {
    expect(wrapper.contains(<h2>Hello World!</h2>)).toEqual(true);
  });

  if (title === "Elephant in the room....") {
    it("renders correctly", () => {
      expect(wrapper.contains(<h2>Elephant in the room....</h2>)).toEqual(true);
    });
  } else {
    it("renders correctly", () => {
      expect(wrapper.contains(<h2 />)).toEqual(false);
    });
  }
});
