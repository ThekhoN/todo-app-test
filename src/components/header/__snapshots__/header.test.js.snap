// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    title="Hello World!"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Hello World!",
    },
    "ref": null,
    "rendered": "Hello World!",
    "type": "h2",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hello World!",
      },
      "ref": null,
      "rendered": "Hello World!",
      "type": "h2",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
